File: C++/A/259A-littleelephantandchess.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    vector<string> board;

public:
    void get_board()
    {
        for (int i = 0; i < 8; i++)
        {
            string x;
            cin >> x;
            board.push_back(x);
        }
    }
    bool validator()
    {
        for (string row : board)
            if (row.find("BB") != string::npos || row.find("WW") != string::npos)
                return false;
        return true;
    }
};

int main()
{
    Solver Object;
    Object.get_board();
    cout << ((Object.validator()) ? "YES" : "NO");
}

File: Python/259A-littleelephantandchess.py
class Solver:
    def __init__(self):
        self.board = []

    def get_board(self):
        for _ in range(8):
            x = input()
            self.board.append(x)

    def validator(self):
        for row in self.board:
            if "BB" in row or "WW" in row:
                return False
        return True


Object = Solver()
Object.get_board()
print("YES" if Object.validator() else "NO")
